File: JavaScript/find_a_palindrome.js
function palindrome(inputString) {

// replace all special characters with blanks and convert to lowercase
    var lettersString = inputString.replace(/[^a-z0-9]/gi, "").toLowerCase();

// split and reverse the string
    var arr = lettersString.split('');
    arr.reverse();
// rejoin as a new string
    var reverseString = arr.join('');
// returns true or false

    console.log(lettersString === reverseString);

    return lettersString === reverseString;
}

palindrome('Aba a_ba');

File: Python/find_a_palindrome.py
import re

def palindrome(input_string):
    # sub all non alphanumeric with ''
    letters_string = re.sub('[^0-9A-Za-z]', '', input_string)
    
    # convert to lowercase
    letters_string = letters_string.lower()
    
    # function to reverse a string
    def reversed_string(aString):
        return aString[::-1]

    rev_string = reversed_string(letters_string)

    print(rev_string)
    print(letters_string)
    print(rev_string == letters_string)

palindrome('A man, a plan, a canal. Panama')
